pnpbridge/src/pnpbridge/src/pnpbridge.c

```diff
@@ -202,13 +191,12 @@ PnpBridge_Main(const char * ConfigurationFilePath)
     } 
 exit:
     {
-        // Destroy the DigitalTwinClient and recreate it
-        IotComms_DigitalTwinClient_Destroy(&pnpBridge->IotHandle);
-        g_PnpBridge = NULL;
-
-        if (pnpBridge) {
+        if (pnpBridge)
+        {
+            IotComms_DeinitializeIotHandle(&pnpBridge->IotHandle, pnpBridge->Configuration.ConnParams);
             PnpBridge_Release(pnpBridge);
```

Comment:
Since adapter maybe reporting telemetry at this time consider 
(1) performing the stop component callbacks 
(2) IotComms_DeinitializeIotHandle
(3) Destroy components 

That way any telemetry that maybe in flight, the adapter has a chance to report it, before it gets taken down.

Reply:
Thank you for catching this :)

---
In reply to: [469417527](https://github.com/Azure/AzurePnPBridgePreview/pull/34#discussion_r469417527) [](ancestors = 469417527)